save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/service/DockerService.kt

```diff
@@ -115,24 +118,34 @@ class DockerService(
             .body(BodyInserters.fromValue(ExecutionUpdateDto(executionId, ExecutionStatus.RUNNING)))
             .retrieve()
             .toBodilessEntity()
+            .map {
+                agentRunner.start(execution.id!!)
+                log.info("Made request to start containers for execution.id=$executionId")
+            }
+            .flatMapMany {
+                // Check, whether the agents were actually started, if yes, all cases will be covered by themselves and HeartBeatInspector,
+                // if no, mark execution as failed with internal error here
+                val now = Clock.System.now()
+                var duration: Long = 0
```

Comment:
`AtomicLong`?

Reply:
But I don't see the actual reason of it
Why do you think, that it is required? `duration` only used in `dockerservice` in this loop, and no other processes will access it

Reply:
Subscription may be executed on a thread pool and it won't guarantee that all operations inside `map` or `flatmap` will be executed on the same thread

Reply:
Make sense, changed